### PR TITLE
Add ID3v2 binary template

### PR DESCRIPTION
**Pull request by wader**

### CODE REVIEW

File: templates/Audio/ID3v2.tcl [hunk 1]
```diff
@@ -0,0 +1,672 @@
...
+        (($n & 0x00007f00) >> 1) | \
+        (($n & 0x0000007f) >> 0) \
+    ]
+    if { $name != "" } {
+	    entry $name [format "%d (%d)" $v $n] 4 [expr [pos]-4]
+    }
+    return $v
+}
+
+proc uint24 { {name ""} } {
```
Comment (wader) [inline templates/Audio/ID3v2.tcl]: Should be a native command?

File: templates/Audio/ID3v2.tcl [hunk 2]
```diff
@@ -0,0 +1,672 @@
...
+        incr i 2
+        if { [uint16] == $v } {
+            break
+        }
+    }
+    move [expr -$i]
+    return [expr $i]
+}
+
+proc ascii_maybe_empty { size {name ""} } {
```
Comment (wader) [inline templates/Audio/ID3v2.tcl]: These maybe-helpers are used because of the size > 0 restriction

File: templates/Audio/ID3v2.tcl [hunk 3]
```diff
@@ -0,0 +1,672 @@
...
+
+proc trim_suffix { suffix s } {
+    set len [string length $suffix]
+    while { [string match "*$suffix" $s] } {
+        set s [string range $s 0 "end-$len"]
+    }
+    return $s
+}
+
+# TODO: some better way of doing this?
```
Comment (wader) [inline templates/Audio/ID3v2.tcl]: Could not figure a bette way of doing this in TCL

**Comment by wader** [comment]
Some other thoughts and suggestions while developing this. Let me know if they should be their own issues:

Im working some other binary templates that like WAV that can ingle ID3v2 tags. How could we make it possible for such a template to include the ID3v2 one? let the WAV template source it somehow? add a native command to do "nested" templates somehow?

I've noticed that if a `entry` command is last in a section it's byte range is not included in the whole sections byte range.

Doing a negative `move` past the start of the file causes weird things to happen.

Better handling when a template fails. Like show last read position etc?

**Comment by wader** [comment]
Thanks for adding binary template support btw! very useful, already used it alot!

File: templates/Audio/ID3v2.tcl (hunk 1 above)
Comment (kainjow) [reply templates/Audio/ID3v2.tcl]: `uint24` is already built-in, but I guess I forgot to document it.

Comment (wader) [reply templates/Audio/ID3v2.tcl]: Aha also didn't noticed. Fixed now.

**Comment by kainjow** [comment]
Yes, please create new topics per issue. Easier to keep track.

**Pull request merged**